plugins/woocommerce/src/Internal/FraudProtection/JetpackConnectionManager.php

```diff
@@ -0,0 +1,227 @@
+			$status['error_code'] = 'jetpack_not_available';
+			return $status;
+		}
+
+		// Get connection manager.
+		$manager = $this->get_connection_manager();
+		if ( ! $manager ) {
+			$status['error']      = __( 'Failed to initialize Jetpack Connection Manager.', 'woocommerce' );
+			$status['error_code'] = 'manager_init_failed';
+			return $status;
+		}
+
+		// Check if connected.
+		if ( ! $manager->is_connected() ) {
+			$status['error']      = __( 'Site is not connected to WordPress.com. Please connect your site to enable fraud protection.', 'woocommerce' );
+			$status['error_code'] = 'not_connected';
+			return $status;
+		}
+
+		// Get blog ID.
+		$blog_id = $this->get_blog_id();
+		if ( ! $blog_id ) {
+			$status['error']      = __( 'Jetpack blog ID not found. Please reconnect your site to WordPress.com.', 'woocommerce' );
+			$status['error_code'] = 'no_blog_id';
+			return $status;
+		}
+
+		// All checks passed.
+		$status['connected'] = true;
+		$status['blog_id']   = $blog_id;
+
+		return $status;
+	}
+
+	/**
+	 * Get the Jetpack authorization URL for connecting the site.
+	 *
+	 * @param string $redirect_url URL to redirect to after authorization.
+	 * @return string|null Authorization URL or null on error.
+	 */
+	public function get_authorization_url( string $redirect_url = '' ): ?string {
+		if ( ! $this->is_jetpack_connection_available() ) {
+			return null;
+		}
+
+		$manager = $this->get_connection_manager();
+		if ( ! $manager ) {
+			return null;
+		}
+
+		// If no redirect URL provided, use current admin URL.
+		if ( empty( $redirect_url ) ) {
+			$redirect_url = admin_url( 'admin.php?page=wc-settings&tab=advanced&section=features' );
+		}
+
+		try {
+			// Use the OnboardingPlugins class to get authorization URL.
+			if ( class_exists( '\Automattic\WooCommerce\Admin\API\OnboardingPlugins' ) ) {
+				$request = new \WP_REST_Request();
+				$request->set_param( 'redirect_url', $redirect_url );
+				$plugin_onboarding = new \Automattic\WooCommerce\Admin\API\OnboardingPlugins();
+				$result            = $plugin_onboarding->get_jetpack_authorization_url( $request );
+
+				if ( ! empty( $result['url'] ) ) {
+					// Customize the URL parameters for fraud protection.
+					$url = add_query_arg(
+						array(
+							'redirect_uri' => $redirect_url,
+							'from'         => 'woocommerce-fraud-protection',
+							'plugin_name'  => 'woocommerce',
+						),
+						$result['url']
+					);
+					return $url;
+				}
+			}
+
+			return null;
+		} catch ( \Exception $e ) {
+			$this->log_error(
+				sprintf(
+					'Failed to get Jetpack authorization URL: %s',
+					$e->getMessage()
+				)
+			);
+			return null;
+		}
+	}
+
+	/**
+	 * Log an error message.
+	 *
+	 * @param string $message Error message.
+	 * @return void
+	 */
+	private function log_error( string $message ): void {
+		$logger = wc_get_logger();
+		$logger->error( $message, array( 'source' => self::LOGGER_SOURCE ) );
+	}
```

Comment:
`FraudProtectionController` has a public `log()` helper. Should we use that instead?